test/gtest/uct/ib/test_rc.cc

```diff
@@ -556,6 +556,59 @@ UCS_TEST_SKIP_COND_P(test_rc_get_limit, ordering_comp_cb,
 
 UCT_INSTANTIATE_RC_DC_TEST_CASE(test_rc_get_limit)
 
+class test_rc_ece_auto : public test_rc {
+public:
+    void init() {
+        m_recv_handler_counter = 0;
+        EXPECT_EQ(UCS_OK, uct_config_modify(m_iface_config, "RC_ECE", "auto"));
+        test_rc::init();
+    }
+
+    static size_t send_pack_cb(void *dest, void *arg)
+    {
+        size_t length = (size_t)arg;
+
+        memset(dest, 0, length);
+        return length;
+    }
+
+     static ucs_status_t
+     recv_handler(void *arg, void *data, size_t length, unsigned flags)
+     {
+         EXPECT_EQ((size_t)arg, length);
+         ++test_rc_ece_auto::m_recv_handler_counter;
+         return UCS_OK;
+     }
+
+    void send_recv(uct_ep_h ep, entity *ent, size_t length)
+    {
+        size_t init_recv_counter = test_rc_ece_auto::m_recv_handler_counter;
+        size_t len;
+
+        /* set a callback for the uct to invoke for receiving the data */
+        uct_iface_set_am_handler(ent->iface(), 0, recv_handler, (void*)length, 0);
+
+        /* send the data */
+        len = uct_ep_am_bcopy(ep, 0, send_pack_cb, (void*)length, 0);
```

Comment:
ssize_t packed_size = uct_ep_am_bcopy(...)
in general we should progress/retry loop as long as the return value is UCS_ERR_NO_RESOURCE